client/X11/xf_rail.c

```diff
@@ -269,90 +303,153 @@ void xf_rail_paint(xfContext* xfc, INT32 uleft, INT32 utop, UINT32 uright,
 
 /* RemoteApp Core Protocol Extension */
 
-static BOOL xf_rail_window_common(rdpContext* context,
-                                  const WINDOW_ORDER_INFO* orderInfo, const WINDOW_STATE_ORDER* windowState)
+static BOOL resizeMarginXSet = FALSE;
```

Comment:
Please do not use static variables for this, add it to the rail context.

Reply:
Ok, no problem. Do you mean the `_rail_client_context` (from `include/freerdp/client/rail.h`)? Or the `xfContext` (aka `xf_context`, from `client/X``/xfreerdp.h`)? Partly why I kept it static was I wasn't sure where it should go.